### PR TITLE
POC: support for k8s secrets

### PR DESCRIPTION
**Pull request by ffais**
This is a POC to resolve this issue #323. I tested on my cluster and works fine.
There're some issue with my implementation, for example secret namespace and secret key are hard coded.

If you have suggestion on how load secret namespace and key from the CR let me know.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -73,6 +74,10 @@ func main() {
 		"GCP project id. Set this value to use cloudSecretManagerType=gcp. "+
 			"Also can be set by environment variable PROJECT_ID."+
 			"If both are set, the flag is used.")
+	flag.StringVar(&secretNamespace, "k8s-secret-namespace", "",
+		"GCP project id. Set this value to use cloudSecretManagerType=gcp. "+
+			"Also can be set by environment variable PROJECT_ID."+
+			"If both are set, the flag is used.")
```
Comment (nakamasato) [inline cmd/main.go:80]: Can you update the description too?

File: cmd/main.go [hunk 2]
```diff
@@ -109,7 +114,8 @@ func main() {
 	secretManagers := map[string]secret.SecretManager{
 		"raw": secret.RawSecretManager{},
 	}
-	if cloudSecretManagerType == "gcp" {
+	switch cloudSecretManagerType {
```
Comment (nakamasato) [inline cmd/main.go:117]: i didn't expect to have k8s secret at the beginning, so cloudSecretManagerType would be not appropriate?
`adminUserSecretType` or sth?

**Review by nakamasato — COMMENTED** [review]
Thank you for your contribution. 
Basically LGTM but left tiny comments!

**Comment by nakamasato** [comment]
Can you also update docs?

File: cmd/main.go (hunk 1 above)
Comment (ffais) [reply cmd/main.go]: Done.

File: cmd/main.go (hunk 2 above)
Comment (ffais) [reply cmd/main.go]: Sounds good to me

**Comment by ffais** [comment]
> Can you also update docs?

I'll work on the documentations in the next days, it will probably also be necessary to update the Helm chart to align it with the documentation.

**Comment by nakamasato** [comment]
@ffais I think we need to grant the get permission of Secret to mysql_controller. Can you add it? Other than that, I think we can merge your pr.
https://github.com/nakamasato/mysql-operator/blob/dc9848bd8105d681030a987222c15a58740ec471/internal/controller/mysql_controller.go#L49-L53

**Comment by nakamasato** [comment]
@ffais I think we need to grant the get permission of Secret to mysql_controller. Can you add it? Other than that, I think we can merge your pr.
https://github.com/nakamasato/mysql-operator/blob/dc9848bd8105d681030a987222c15a58740ec471/internal/controller/mysql_controller.go#L49-L53

File: internal/controller/mysqluser_controller.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ type MySQLUserReconciler struct {
 //+kubebuilder:rbac:groups=mysql.nakamasato.com,resources=mysqlusers,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=mysql.nakamasato.com,resources=mysqlusers/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=mysql.nakamasato.com,resources=mysqlusers/finalizers,verbs=update
-//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
```
Comment (nakamasato) [inline internal/controller/mysqluser_controller.go:64]:
```suggestion
//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create
```

`create` is necessary for `MySQLUserController`.

Comment (ffais) [reply internal/controller/mysqluser_controller.go]: Done!

File: config/rbac/role.yaml [hunk 4]
```diff
@@ -9,12 +9,8 @@ rules:
   resources:
   - secrets
   verbs:
```
Comment (nakamasato) [inline config/rbac/role.yaml:11]:
```suggestion
  verbs:
    - create
```

Comment (nakamasato) [reply config/rbac/role.yaml:11]: I think the annotation is not reflected to the rbac!

**Review by nakamasato — APPROVED** [review]
Thank you for your contribution! 
I've updated config/rbac/role.yaml and all CI passed!

**Pull request merged**